lib/trie/hash_test.go

```diff
@@ -7,86 +7,69 @@ import (
 	"bytes"
 	"math/rand"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func generateRandBytes(size int) []byte {
-	r := *rand.New(rand.NewSource(rand.Int63()))
-	buf := make([]byte, r.Intn(size)+1)
-	r.Read(buf)
+	buf := make([]byte, rand.Intn(size)+1)
+	rand.Read(buf)
 	return buf
 }
 
 func generateRand(size int) [][]byte {
 	rt := make([][]byte, size)
-	r := *rand.New(rand.NewSource(rand.Int63()))
 	for i := range rt {
-		buf := make([]byte, r.Intn(379)+1)
-		r.Read(buf)
+		buf := make([]byte, rand.Intn(379)+1)
+		rand.Read(buf)
 		rt[i] = buf
 	}
 	return rt
 }
 
-func TestNewHasher(t *testing.T) {
-	hasher := newHasher(false)
-	defer hasher.returnToPool()
-
-	_, err := hasher.hash.Write([]byte("noot"))
-	if err != nil {
-		t.Error(err)
-	}
-
-	sum := hasher.hash.Sum(nil)
-	if sum == nil {
-		t.Error("did not sum hash")
-	}
-
-	hasher.hash.Reset()
-}
-
 func TestHashLeaf(t *testing.T) {
-	hasher := newHasher(false)
-	defer hasher.returnToPool()
-
 	n := &leaf{key: generateRandBytes(380), value: generateRandBytes(64)}
-	h, err := hasher.Hash(n)
+
+	buffer := bytes.NewBuffer(nil)
+	const parallel = false
+	err := encodeNode(n, buffer, parallel)
+
 	if err != nil {
 		t.Errorf("did not hash leaf node: %s", err)
-	} else if h == nil {
+	} else if buffer.Len() == 0 {
```

Comment:
could we change to `require.NoError` and `require.Greater`?

Reply:
If you don't mind I'm working on refactoring tests and existing code in another branch, to keep deltas low in this already decently large PR :smile: But definitely something to do!

Reply:
Done in https://github.com/ChainSafe/gossamer/pull/2049